promtsdb/promtsdb.go

```diff
@@ -71,26 +85,32 @@ func (q *V3ioPromQuerier) Select(params *storage.SelectParams, oms ...*labels.Ma
 			}
 		}
 	}
-	set, err := q.q.SelectProm(name, functions, params.Step, filter, noAggr)
+	set, err := promQuery.v3ioQuerier.SelectProm(name, functions, params.Step, filter, noAggr)
 	return &V3ioPromSeriesSet{s: set}, err
 }
 
 // LabelValues returns all potential values for a label name.
-func (q *V3ioPromQuerier) LabelValues(name string) ([]string, error) {
-	return q.q.LabelValues(name)
+func (promQuery *V3ioPromQuerier) LabelValues(name string) ([]string, error) {
+	return promQuery.v3ioQuerier.LabelValues(name)
 }
 
 // Close releases the resources of the Querier.
-func (q *V3ioPromQuerier) Close() error {
+func (promQuery *V3ioPromQuerier) Close() error {
```

Comment:
It's actually somewhat of a convention to abbreviate the receiver :)

Reply:
it was confusing because of another `q` but I can rename it to `vpq` if you wish